client/checkout/classic/index.js

```diff
@@ -0,0 +1,196 @@
+		hidePostalCode: true,
+		classes: { base: 'wcpay-card-mounted' },
+	} );
+
+	// Only attempt to mount the card element once that section of the page has loaded. We can use the updated_checkout
+	// event for this. This part of the page can also reload based on changes to checkout details, so we call unmount
+	// first to ensure the card element is re-mounted correctly.
+	$( document.body ).on( 'updated_checkout', function() {
+		// Don't re-mount if already mounted in DOM.
+		if ( $( '#wcpay-card-element' ).children().length ) {
+			return;
+		}
+
+		cardElement.unmount();
+		cardElement.mount( '#wcpay-card-element' );
+	} );
+
+	if ( $( 'form#add_payment_method' ).length || $( 'form#order_review' ).length ) {
+		cardElement.mount( '#wcpay-card-element' );
+	}
+
+	// Update the validation state based on the element's state.
+	cardElement.addEventListener( 'change', function( event ) {
+		const displayError = $( '#wcpay-errors' );
+		if ( event.error ) {
+			displayError.html( '<ul class="woocommerce-error"><li /></ul>' )
+				.find( 'li' ).text( event.error.message );
+		} else {
+			displayError.empty();
+		}
+	} );
+
+	// Show error notice at top of checkout form.
+	const showError = ( errorMessage ) => {
+		const messageWrapper = '<ul class="woocommerce-error" role="alert">' + errorMessage + '</ul>';
+		const $container = $( '.woocommerce-notices-wrapper, form.checkout' ).first();
+
+		if ( ! $container.length ) {
+			return;
+		}
+
+		// Adapted from WooCommerce core @ ea9aa8c, assets/js/frontend/checkout.js#L514-L529
+		$( '.woocommerce-NoticeGroup-checkout, .woocommerce-error, .woocommerce-message' ).remove();
+		$container.prepend( '<div class="woocommerce-NoticeGroup woocommerce-NoticeGroup-checkout">' + messageWrapper + '</div>' );
+		$container.find( '.input-text, select, input:checkbox' ).trigger( 'validate' ).blur();
+
+		let scrollElement = $( '.woocommerce-NoticeGroup-checkout' );
+		if ( ! scrollElement.length ) {
+			scrollElement = $container;
+		}
+
+		$.scroll_to_notices( scrollElement );
+		$( document.body ).trigger( 'checkout_error' );
+	};
+
+	// Create payment method on submission.
+	let paymentMethodGenerated;
+
+	/**
+	 * Generates a payment method, saves its ID in a hidden input, and re-submits the form.
+	 *
+	 * @param {object} $form The jQuery object for the form.
+	 * @return {boolean} A flag for the event handler.
+	 */
+	const handleOnPaymentFormSubmit = ( $form ) => {
+		// We'll resubmit the form after populating our payment method, so if this is the second time this event
+		// is firing we should let the form submission happen.
+		if ( paymentMethodGenerated ) {
+			paymentMethodGenerated = null;
+			return;
+		}
+
+		// Block UI to indicate processing and avoid duplicate submission.
+		$form.addClass( 'processing' ).block( {
+			message: null,
+			overlayCSS: {
+				background: '#fff',
+				opacity: 0.6,
+			},
+		} );
+
+		const request = api.generatePaymentMethodFromCard( {
```

Comment:
Looks like this needs updating (now `generatePaymentMethodRequest`?). I get `generatePaymentMethodRequest is not a function` error when using WC Pay in classic (shortcode) checkout.

Reply:
Oh, my bad. I renamed the method and one of the instances where it's used, but not the other one. Fixed in ce8919d